src/psyclone/psyGen.py

```diff
@@ -415,10 +415,13 @@ class Invoke(object):
     ''' Manage an individual invoke call '''
 
     def __str__(self):
-        return self._name+"("+str(self.unique_args)+")"
+        return self._name+"("+", ".join([str(arg) for arg in
+                                         self._alg_unique_args])+")"
 
     def __init__(self, alg_invocation, idx, schedule_class,
                  reserved_names=None):
```

Comment:
Please could you make a start on documenting the interface to this constructor. If you don't know what all the arguments are then please just leave placeholders.

Reply:
OK, I tried :)